lib/active_merchant/billing/gateways/braintree/token_nonce.rb

```diff
@@ -35,8 +35,13 @@ def create_token_nonce_for_payment_method(payment_method)
       end
 
       def client_token
-        base64_token = @braintree_gateway.client_token.generate
-        JSON.parse(Base64.decode64(base64_token))['authorizationFingerprint']
+        parameters = options[:merchant_account_id] ? { merchant_account_id: options[:merchant_account_id] } : {}
+        base64_token = @braintree_gateway.client_token.generate(parameters)
+        JSON.parse(Base64.decode64(base64_token))
+      end
+
+      def authorization_fingerprint
```

Comment:
❓  Question:

I get the point of calling it `authorization_fingerprint` instead of `client_token` but a method just get a key from a hash ?

Reply:
Make sense, I will remove `authorization_fingerprint`